docs/userguide/mpi_apps.rst

```diff
@@ -0,0 +1,154 @@
+MPI Apps
+========
+
+.. note::
+
+    Parsl support for MPI Apps is being re-engineered.
+    We describe the best practices with today's Parsl.
+    Join our Slack if you want to steer Parsl's future.
+
+MPI applications run multiple copies of a program that complete a single task by
+coordinating using messages passed within or across nodes.
+Starting MPI application requires invoking a "launcher" code (e.g., ``mpiexec``) from one node
+with options that define how the copies of a program should be distributed to others.
+The need to call a launcher from a specific node requires a special configuration of Parsl `Apps <apps.html>`_
+and `Execution <execution.html>`_ environments to run Apps which use MPI.
+
+HTEx and MPI Tasks
+------------------
+
+The :class:`~parsl.executors.high_throughput.executor.HighThroughputExecutor` (HTEx) is the default execution provider
+available through Parsl.
+Parsl Apps which invoke MPI code require a dedicated HTEx configured such that every Parsl app
+will have access to any of the nodes available within a block,
+and Apps that will invoke the MPI launcher with the correct settings.
+
+Configuring the Provider
+++++++++++++++++++++++++
+
+Parsl must be configured to deploy exactly one executor per execution block.
+This part is simple.
+Instead of defining `a launcher <execution.html#launchers>`_ which will
+place an executor on each node in the block, simply use the :class:`~parsl.launchers.SimpleLauncher`.
+
+It is also necessary to specify the desired number of blocks for the executor.
+Parsl cannot determine the number of blocks needed to run a set of MPI Apps,
```

Comment:
does this document ever assume you'll be running a single MPI-app-instance (phrasing?) across nodes provisioned from several blocks? i think not?

Reply:
No, it does not. I'll add a note about blocks being a hard boundary for MPI applications to the "Limitations"